### PR TITLE
QOL improvements for menuItem's icons and Images

### PR DESCRIPTION
**Pull request by HookedBehemoth**
These changes are purely for scaling and rescaling.
MenuItem's icon will be scaled and centered while images will be rescaled when changing the icon(-path).

### CODE REVIEW

File: Plutonium/Source/pu/ui/elm/elm_Menu.cpp [hunk 1]
```diff
@@ -293,11 +293,20 @@ namespace pu::ui::elm
...
-                    s32 icd = (this->isize - 10);
+                    float factor = (float)render::GetTextureHeight(curicon)/(float)render::GetTextureWidth(curicon);
+                    s32 icw = (this->isize - 10);
+                    s32 ich = icw;
                     s32 icx = (cx + 25);
                     s32 icy = (cy + 5);
-                    tx = (icx + icd + 25);
-                    Drawer->RenderTextureScaled(curicon, icx, icy, icd, icd);
+                    tx = (icx + icw + 25);
+                    if(factor < 1){
```
Comment (XorTroll) [inline Plutonium/Source/pu/ui/elm/elm_Menu.cpp]: Mind following Plutonium's codestyle, basically using a new line to start the if-else statements' body?

File: Plutonium/Source/pu/ui/elm/elm_Menu.cpp [hunk 2]
```diff
@@ -293,11 +293,20 @@ namespace pu::ui::elm
...
+                        ich = ich*factor;
+                        icy = icy+((this->isize-ich)/2);
+                    } else {
```
Comment (XorTroll) [inline Plutonium/Source/pu/ui/elm/elm_Menu.cpp]: Same here ^

Comment (HookedBehemoth) [reply Plutonium/Source/pu/ui/elm/elm_Menu.cpp]: fixed

**Comment by HookedBehemoth** [comment]
Oh shit
That last commit should not have been added to the PR

**Comment by HookedBehemoth** [comment]
panic mode activated

**Comment by HookedBehemoth** [comment]
tested PR

**Comment by HookedBehemoth** [comment]
BTW, this is still up to date and works on the latest commit

**Pull request merged**